eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -975,10 +981,13 @@ protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId)
             $row['id'] = $row['link'] = $newIdsMap[$oldParentAliasId];
             $this->gateway->insertRow($row);
 
+            $alreadyGeneratedPairs[] = $temporaryRowIdentifier;
```

Comment:
Depending on the amount of entries (I assume copying a subtree might involve copying **a lot**) that might be generated it might be sensible to use a hash-map instead, making checks `o(1)` in terms of complexity (instead of `o(n)` in current variant)
```suggestion
            $alreadyGeneratedPairs[$temporaryRowIdentifier] = true;
```
Then you can check if identifier was already generated by:
```php
isset($alreadyGeneratedPairs[$identifier]);
```
which would replace your `in_array` call.